scala-core-3/src/main/scala/com/baeldung/scala/withfilter/WithFilterVsFilter.scala

```diff
@@ -0,0 +1,56 @@
+package com.baeldung.scala.withfilter
+
+object WithFilterVsFilter extends App {
+  sealed trait Level
+  object Level {
+    case object Junior extends Level
+    case object Mid extends Level
+    case object Senior extends Level
+  }
+  case class Programmer(name: String,
+                        level: Level,
+                        knownLanguages: List[String])
+
+  val programmers: List[Programmer] = List(
+    Programmer(name = "Kelly",
+               level = Level.Mid,
+               knownLanguages = List("JavaScript")),
+    Programmer(name = "John",
+               level = Level.Senior,
+               knownLanguages = List("Java", "Scala", "Kotlin")),
+    Programmer(name = "Dave",
+               level = Level.Junior,
+               knownLanguages = List("C", "C++"))
+  )
+
+  val getName: Programmer => String =
+    programmer => {
+      println("get name" + programmer)
```

Comment:
Do we want to use console output here? or should we be using unit tests to demonstrate what the code does?